net/instaweb/rewriter/add_instrumentation.js

```diff
@@ -95,6 +95,12 @@ pagespeed.AddInstrumentation.prototype.sendBeacon = function() {
     if (timingApi['msFirstPaint']) {
       // IE.
       firstPaintTime = timingApi['msFirstPaint'];
+    } else if (window['performance']['timeOrigin'] &&
+        window['performance']['getEntriesByName']('first-paint').length > 0) {
+        // Paint Timing API
+        firstPaintTime = Math.floor(
+            window.performance.timeOrigin +
+            window.performance.getEntriesByName('first-paint')[0].startTime);
```

Comment:
I'm a bit OCD when it comes to CSE :)  WDYT of making a temp `winPerf` for window['performance'] and then using it afterward, and likeways for winPerf['getEntriesByName'] and others where you test first and then call, or have chains of dereferencing...

This is just a suggestion, and it would require something like a nested helper function to help you manage the temps in the conditional.

Reply:
I'm not sure how I would use a helper function for preventing chains of dereferencing, could you clarify that bit?
I gave it a shot with the temporary variables, but feel like there's a better way of doing it.